scribus/text/storytext.cpp

```diff
@@ -252,21 +254,22 @@ void StoryText::moveCursorRight(bool isGrapheme)
 {
 	if (isGrapheme)
 	{
-		int pos;
-		QTextBoundaryFinder boundaryFinder(QTextBoundaryFinder::Grapheme, plainText());
+		BreakIterator* it = getGraphemeIterator();
+		if (!it)
+			return;
+
+		int pos = cursorPosition();
+		it->setText(plainText().utf16());
 		if (paragraphStyle().direction() == ParagraphStyle::RTL)
 		{
-			boundaryFinder.setPosition(cursorPosition() - 1);
-			for (pos = -1; !boundaryFinder.isAtBoundary(); pos--)
-				boundaryFinder.setPosition(boundaryFinder.position() - 1);
+			pos = cursorPosition();
```

Comment:
After removing this line, you can remove the curly brackets around the blocks as they will be redundant.